pkg/validate/tree/hydrate/namespace_selector_hydrator.go

```diff
@@ -59,6 +59,13 @@ func labelSelector(obj ast.FileObject) (labels.Selector, status.Error) {
 	}
 	nss := s.(*v1.NamespaceSelector)
 
+	if nss.Spec.Mode == v1.NSSelectorDynamicMode {
+		return nil, selectors.UnsupportedNamespaceSelectorModeError(nss)
+	}
+	if nss.Spec.Mode != "" && nss.Spec.Mode != v1.NSSelectorStaticMode {
+		return nil, selectors.UnknownNamespaceSelectorModeError(nss)
+	}
```

Comment:
Can we just return `InvalidSelectorError(obj, err)` here (combined with below), and `err` tells users the cause, e.g. "dynamic mode isn't supported in hierachical repositories" etc.

Reply:
Updated to reuse KNV1014, but created separate functions to wrap the error message, which seems the common way to build error message.